can/interfaces/socketcan/socketcan_ctypes.py

```diff
@@ -88,18 +77,13 @@ def set_filters(self, can_filters=None):
             A filter matches, when ``<received_can_id> & can_mask == can_id & can_mask``
 
         """
-        if can_filters is not None and len(can_filters) > 0:
-            can_filter_fmt = "={}I".format(2 * len(can_filters))
-            filter_data = []
-            for can_filter in can_filters:
-                filter_data.append(can_filter['can_id'])
-                filter_data.append(can_filter['can_mask'])
-            res = libc.setsockopt(self.socket, SOL_CAN_RAW,
-                CAN_RAW_FILTER, struct.pack(can_filter_fmt, *filter_data),
-                4*2*len(kwargs['can_filters'])
-                )
-            if res != 0:
-                print(str(res))
+        can_filter_fmt, filter_data = parseCanFilters(can_filters)
+        res = libc.setsockopt(self.socket, SOL_CAN_RAW,
+                              CAN_RAW_FILTER, struct.pack(can_filter_fmt, *filter_data),
+                              len(filter_data)*ctypes.sizeof(ctypes.c_uint32)
+                             )
+        if res != 0:
+            log.debug('Setting filters failed: ' + str(res))
```

Comment:
This should be a much higher priority log message (e.g. `log.error`).

I'm tempted to say we should also raise a `CanError` exception. Have you seen it fail, do you know when it might?